src/core/Akka.Remote/Transport/Helios/HeliosTcpTransport.cs

```diff
@@ -239,12 +186,30 @@ public HeliosTcpTransport(ActorSystem system, Config config)
 
         protected override Task<AssociationHandle> AssociateInternal(Address remoteAddress)
         {
-            var client = NewClient(remoteAddress);
+            var clientBootstrap = ClientFactory(remoteAddress);
+            var socketAddress = AddressToSocketAddress(remoteAddress);
+            var associate = clientBootstrap.ConnectAsync(socketAddress).ContinueWith(tr =>
+            {
+                var channel = tr.Result;
+                var handler = (TcpClientHandler) channel.Pipeline.Last();
```

Comment:
the `IChannel.Pipeline` is safe to access in this context since it's never modified after startup.